include/highfive/bits/H5Easy_vector.hpp

```diff
@@ -53,6 +53,23 @@ struct overwrite_impl
         return dataset;
     }
 };
+    
+// load "std::vector<std::vector>" from DataSet
+template <class T>
+struct load_impl<std::vector<std::vector<T>>>
+{
+    static std::vector<std::vector<T>> run(const File& file, const std::string& path)
+    {
+        DataSet dataset = file.getDataSet(path);
+        std::vector<size_t> dims = dataset.getDimensions();
+        if (dims.size() != 2) {
```

Comment:
In fact your addition is entirely obsolete. If the dimension check is removed or made more intelligent in the existing overload the code will readily work (`std::vector<std::vector<double>>` is a valid specialisation of `std::vector<T>`).

What surprises me a little however, it that I would expect the same for the dump overload, but that one does seem to require an explicit `std::vector<std::vector<T>>` overload.